src/generate/peripheral.rs

```diff
@@ -323,13 +323,13 @@ impl FieldRegions {
         let mut indices = Vec::new();
 
         let field_start = field.offset;
-        let field_end = field_start + field.size / BITS_PER_BYTE;
+        let field_end = field_start + (( field.size as f32 ) / (BITS_PER_BYTE as f32)).ceil() as u32;
```

Comment:
The more obvious way to achieve the same would be

```suggestion
        let field_end = field_start + (field.size + BITS_PER_BYTE - 1) / BITS_PER_BYTE;
```